### PR TITLE
Upgrade prometheus to 7309ac272195cb856b879306d6a27af7641d3346

### PR DESCRIPTION
**Pull request by alexqyle**
* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes
* Upgrade prometheus to 7309ac272195cb856b879306d6a27af7641d3346
* Upgrade promql-engine to c2396ed5afd94f4fec661bb4b3497b3aad0c2bf3

## Verification
* Run test locally

### CODE REVIEW

File: pkg/reloader/reloader.go [hunk 1]
```diff
@@ -603,6 +604,7 @@ func (w *watcher) run(ctx context.Context) {
 			return
 
 		case event := <-w.w.Events:
+			fmt.Printf("event: %s\n", event.Name)
```
Comment (yeya24) [inline pkg/reloader/reloader.go:607]: Is this needed?

File: pkg/store/storepb/prompb/samples.go [hunk 2]
```diff
@@ -42,6 +42,23 @@ func SamplesFromPromqlPoints(samples ...promql.Point) ([]Sample, []Histogram) {
 	return floats, histograms
 }
 
+// SamplesFromPromqlSeries converts promql.Series to a slice of Sample.
```
Comment (yeya24) [inline pkg/store/storepb/prompb/samples.go:45]: Convert to both samples and histograms. Would you mind updating the comment a little bit

**Review by yeya24 — APPROVED** [review]
Great work! Thanks, some nit only.

**Review by yeya24 — DISMISSED** [review]

**Review by fpetkovski — APPROVED** [review]
Thanks 👍

**Review by GiedriusS — APPROVED** [review]

**Pull request merged**